### ISSUE TITLE
v0.3.15 error darwin x64

### ISSUE THREAD

**Issue by tong467** [microsoft/vscode-java-installer#244]
:warning:Please attach **$TMPDIR/vscode-java-installer/log.txt** for more details.
version: 0.3.15
env:
```
JAVA_HOME: undefined
PATH:
/usr/bin
/bin
/usr/sbin
/sbin
JDKs:
/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home	DefaultLocation
```
log:
```
[上午11:00:32] AdoptOpenJDK-11-x64
[上午11:02:28] RequestError: Error: read ETIMEDOUT
[上午11:02:28] Retry downloading ... [1]
[上午11:05:46] Error: read ECONNRESET
[上午11:05:46] Retry downloading ... [2]
[上午11:07:12] RequestError: Error: read ETIMEDOUT
[上午11:07:12] fallback to Adopt API v3
[上午11:08:41] Error: read ETIMEDOUT
[上午11:08:41] Retry downloading ... [1]
[上午11:09:34] Download JDK ... [Completed]
[上午11:09:34] Install JDK
[上午11:09:45] Error occurred
[上午11:09:45] Error: Command failed: installer -pkg /var/folders/00/23hgpnl172zchv_p0lqk5y000000gn/T/vscode-java-installer/binaries/jdk/11/x64/AdoptOpenJDK-11.pkg -target /


```

**Comment by tong467** [comment]
info: Started {"timestamp":"2021-09-24T02:23:04.310Z"}
verbose: Ids: {"uiKind":1,"appRoot":"/private/var/folders/00/23hgpnl172zchv_p0lqk5y000000gn/T/AppTranslocation/1C8DD4B6-702D-40F9-9A82-4B3A3E51D5B3/d/Coding Pack for Java.app/Contents/Resources/app.asar","isTelemetryEnabled":true,"machineId":"6c9d2bc8f91b89624add29c0abeae7fb42bf539fa1cdb2e3e57cd668fa9bcead","sessionId":"024bfa35-3ca3-4a5e-848a-c67f9324a5d01632450184187","timestamp":"2021-09-24T02:23:04.314Z"}
verbose: Packed binaries: {"timestamp":"2021-09-24T02:23:08.154Z"}
verbose: Found JDKs in env.JAVA_HOME {"jdks":[],"timestamp":"2021-09-24T02:23:08.157Z"}
verbose: Found JDK {"timestamp":"2021-09-24T02:23:08.160Z"}
verbose: Found JDK {"timestamp":"2021-09-24T02:23:08.163Z"}
verbose: Found JDKs in env.PATH {"jdks":["/usr","/usr"],"timestamp":"2021-09-24T02:23:08.165Z"}
verbose: Found JDKs in WindowsRegistry {"jdks":[],"timestamp":"2021-09-24T02:23:08.166Z"}
verbose: Found JDKs in DefaultLocation {"jdks":["/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home"],"timestamp":"2021-09-24T02:23:08.171Z"}
error: release file not found {"javaHome":"/usr","timestamp":"2021-09-24T02:23:08.171Z"}
warn: Unknown version of JDK /usr {"timestamp":"2021-09-24T02:23:08.171Z"}
verbose: java_home candidate: {"candidate":{"home":"/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home","sources":["DefaultLocation"],"version":8},"timestamp":"2021-09-24T02:23:08.174Z"}
verbose: Installed components: {"vscode":{"homedir":"/Applications/Visual Studio Code.app","version":"1.60.2"},"jdk":{"name":"JDK","homedir":"/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home","version":"8"},"extensions":[],"timestamp":"2021-09-24T02:23:08.176Z"}
verbose: jdk metadata {"release_name":"jdk-11.0.12+7","release_link":"https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7","timestamp":"2021-08-03T10:21:50Z","release":true,"binaries":[{"os":"mac","architecture":"x64","binary_type":"jdk","openjdk_impl":"hotspot","binary_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_size":191257594,"checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz.sha256.txt","installer_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_size":191643541,"installer_checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg.sha256.txt","installer_download_count":67239,"version":"11","version_data":{"openjdk_version":"11.0.12+7","semver":"11.0.12+7"},"heap_size":"normal","download_count":87275,"updated_at":"2021-08-01T13:50:48Z"}],"download_count":724718}
info: Started {"timestamp":"2021-09-24T03:00:27.582Z"}
verbose: Ids: {"uiKind":1,"appRoot":"/private/var/folders/00/23hgpnl172zchv_p0lqk5y000000gn/T/AppTranslocation/387D35DC-CDD7-489D-AAF7-376D3EF53432/d/Coding Pack for Java.app/Contents/Resources/app.asar","isTelemetryEnabled":true,"machineId":"6c9d2bc8f91b89624add29c0abeae7fb42bf539fa1cdb2e3e57cd668fa9bcead","sessionId":"386da24d-50a2-4a0b-b634-9ead242b30fe1632452427406","timestamp":"2021-09-24T03:00:27.585Z"}
verbose: Packed binaries: {"timestamp":"2021-09-24T03:00:30.734Z"}
verbose: Found JDKs in env.JAVA_HOME {"jdks":[],"timestamp":"2021-09-24T03:00:30.736Z"}
verbose: Found JDK {"timestamp":"2021-09-24T03:00:30.740Z"}
verbose: Found JDK {"timestamp":"2021-09-24T03:00:30.742Z"}
verbose: Found JDKs in env.PATH {"jdks":["/usr","/usr"],"timestamp":"2021-09-24T03:00:30.743Z"}
verbose: Found JDKs in WindowsRegistry {"jdks":[],"timestamp":"2021-09-24T03:00:30.744Z"}
verbose: Found JDKs in DefaultLocation {"jdks":["/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home"],"timestamp":"2021-09-24T03:00:30.746Z"}
error: release file not found {"javaHome":"/usr","timestamp":"2021-09-24T03:00:30.747Z"}
warn: Unknown version of JDK /usr {"timestamp":"2021-09-24T03:00:30.747Z"}
verbose: java_home candidate: {"candidate":{"home":"/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home","sources":["DefaultLocation"],"version":8},"timestamp":"2021-09-24T03:00:30.753Z"}
verbose: Installed components: {"vscode":{"homedir":"/Applications/Visual Studio Code.app","version":"1.60.2"},"jdk":{"name":"JDK","homedir":"/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home","version":"8"},"extensions":[],"timestamp":"2021-09-24T03:00:30.762Z"}
verbose: jdk metadata {"release_name":"jdk-11.0.12+7","release_link":"https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7","timestamp":"2021-08-03T10:21:50Z","release":true,"binaries":[{"os":"mac","architecture":"x64","binary_type":"jdk","openjdk_impl":"hotspot","binary_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_size":191257594,"checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz.sha256.txt","installer_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_size":191643541,"installer_checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg.sha256.txt","installer_download_count":67239,"version":"11","version_data":{"openjdk_version":"11.0.12+7","semver":"11.0.12+7"},"heap_size":"normal","download_count":87275,"updated_at":"2021-08-01T13:50:48Z"}],"download_count":724718}
verbose: jdk metadata {"release_name":"jdk-11.0.12+7","release_link":"https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7","timestamp":"2021-08-03T10:21:50Z","release":true,"binaries":[{"os":"mac","architecture":"x64","binary_type":"jdk","openjdk_impl":"hotspot","binary_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_size":191257594,"checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz.sha256.txt","installer_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_size":191643541,"installer_checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg.sha256.txt","installer_download_count":67239,"version":"11","version_data":{"openjdk_version":"11.0.12+7","semver":"11.0.12+7"},"heap_size":"normal","download_count":87275,"updated_at":"2021-08-01T13:50:48Z"}],"download_count":724718}
verbose: jdk metadata {"release_name":"jdk-11.0.12+7","release_link":"https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7","timestamp":"2021-08-03T10:21:50Z","release":true,"binaries":[{"os":"mac","architecture":"x64","binary_type":"jdk","openjdk_impl":"hotspot","binary_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz","binary_size":191257594,"checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.tar.gz.sha256.txt","installer_name":"OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg","installer_size":191643541,"installer_checksum_link":"https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg.sha256.txt","installer_download_count":67239,"version":"11","version_data":{"openjdk_version":"11.0.12+7","semver":"11.0.12+7"},"heap_size":"normal","download_count":87275,"updated_at":"2021-08-01T13:50:48Z"}],"download_count":724718}
info: JDK downloaded. {"timestamp":"2021-09-24T03:09:34.897Z"}
error: Command failed: installer -pkg /var/folders/00/23hgpnl172zchv_p0lqk5y000000gn/T/vscode-java-installer/binaries/jdk/11/x64/AdoptOpenJDK-11.pkg -target /
 {"code":1,"timestamp":"2021-09-24T03:09:45.870Z"}
verbose: Found JDKs in env.JAVA_HOME {"jdks":[],"timestamp":"2021-09-24T03:12:07.861Z"}
verbose: Found JDK {"timestamp":"2021-09-24T03:12:07.882Z"}
verbose: Found JDK {"timestamp":"2021-09-24T03:12:07.886Z"}
verbose: Found JDKs in env.PATH {"jdks":["/usr","/usr"],"timestamp":"2021-09-24T03:12:07.887Z"}
verbose: Found JDKs in WindowsRegistry {"jdks":[],"timestamp":"2021-09-24T03:12:07.888Z"}
verbose: Found JDKs in DefaultLocation {"jdks":["/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home"],"timestamp":"2021-09-24T03:12:07.889Z"}
error: release file not found {"javaHome":"/usr","timestamp":"2021-09-24T03:12:07.890Z"}
warn: Unknown version of JDK /usr {"timestamp":"2021-09-24T03:12:07.891Z"}

**Comment by Eskibear** [comment]
VS Code Java tools require JDK 11+ to launch, and it's possible that AdoptOpenJDK installer was broken due to networking issue. 
Can you check file hash of  `/var/folders/00/23hgpnl172zchv_p0lqk5y000000gn/T/vscode-java-installer/binaries/jdk/11/x64/AdoptOpenJDK-11.pkg`?

Workaround is to download and install a JDK 11 on your own.

**Comment by Eskibear** [comment]
Root cause found. AdoptOpenJDK API v3 changed. 
E.g. `https://api.adoptopenjdk.net/v3/binary/latest/11/ga/mac/x64/jdk/hotspot/normal/adoptopenjdk` now points to a `.tar.gz` file instead of `.pkg`

**Comment by Eskibear** [comment]
fixed in 0.3.16
